main.client.ts

```diff
@@ -6,32 +6,35 @@ if (mainNav) {
     "#current-nav-item",
   ) as HTMLElement;
 
-  mainNav.addEventListener(
-    "mouseover",
-    (e) =>
-      requestAnimationFrame(() => {
+  if (currentNavItem && currentNavItemHighlighter) {
+    mainNav.addEventListener(
+      "mouseover",
+      (e) => {
         if (e.target === e.currentTarget) return;
-        const target = e.target as HTMLElement;
-        const defaultLeft = currentNavItem?.getBoundingClientRect().left!;
-        const defaultScaleX = currentNavItem?.getBoundingClientRect().width!;
-        if (e.target === currentNavItem) {
-          currentNavItemHighlighter.style.setProperty("--left", "0px");
-          currentNavItemHighlighter.style.setProperty("--scaleX", "1");
-        } else {
-          currentNavItemHighlighter.style.setProperty(
-            "--left",
-            `${target.getBoundingClientRect().left - defaultLeft}px`,
-          );
-          currentNavItemHighlighter.style.setProperty(
-            "--scaleX",
-            `${target.offsetWidth / defaultScaleX}`,
-          );
-        }
-      }),
-  );
+        requestAnimationFrame(() => {
+          const target = e.target as HTMLElement;
+          const defaultLeft = currentNavItem?.getBoundingClientRect()?.left;
+          const defaultScaleX = currentNavItem?.getBoundingClientRect()?.width;
+          if (target === currentNavItem) {
+            currentNavItemHighlighter.style.setProperty("--left", "0px");
+            currentNavItemHighlighter.style.setProperty("--scaleX", "1");
+          } else if (currentNavItem && defaultLeft && defaultScaleX) {
+            currentNavItemHighlighter.style.setProperty(
+              "--left",
+              `${target.getBoundingClientRect().left - defaultLeft}px`,
+            );
+            currentNavItemHighlighter.style.setProperty(
+              "--scaleX",
+              `${target.offsetWidth / defaultScaleX}`,
+            );
+          }
+        });
+      },
+    );
 
-  mainNav.addEventListener("mouseout", () => {
-    currentNavItemHighlighter.style.setProperty("--left", "0px");
-    currentNavItemHighlighter.style.setProperty("--scaleX", "1");
-  });
+    mainNav.addEventListener("mouseout", () => {
+      currentNavItemHighlighter.style.setProperty("--left", "0px");
+      currentNavItemHighlighter.style.setProperty("--scaleX", "1");
+    });
+  }
```

Comment:
Most of this is just whitespace changes. It just make sure the hover effect works right, and also, doesn't cause console errors on pages without the "current page" marker (e.g., the 404 page)